spec/index.bs

```diff
@@ -8403,50 +8404,58 @@ a list of vertices to process for each instance.
 <div algorithm>
     <dfn abstract-op>resolve indices</dfn>(drawCall, state)
 
-        **Arguments:**
-            - |drawCall|: The draw call parameters.
-            - |state|: The active [=RenderState=].
-
-        **Returns:** list of integer indices.
-
-        1. Let |vertexIndexList| be an empty list of indices.
-        1. If |drawCall| is an indexed draw call:
-            1. initialize the |vertexIndexList| with |drawCall|.indexCount integers.
-            1. for |i| in range 0 .. |drawCall|.indexCount (non-inclusive):
-                1. let |vertexIndex| be [$fetch index$](|i| + |drawCall|.firstIndex,
-                    |state|.[=RenderState/indexBuffer=].buffer, |state|.[=RenderState/indexBuffer=].offset,
-                    |state|.[=RenderState/indexBuffer=].format) + |drawCall|.baseVertex
-                1. append |vertexIndex| to the |vertexIndexList|
-        1. Otherwise:
-            1. initialize the |vertexIndexList| with |drawCall|.vertexCount integers.
-            1. assign the |vertexIndexList| item |i| to be |drawCall|.firstVertex + |i|
-        1. Return |vertexIndexList|.
-
-        Note: in case of indirect draw calls, the `indexCount`, `vertexCount`,
-        and other properties of |drawCall| are read from the indirect buffer
-        instead of the draw command itself.
-
-        Issue: specify indirect commands better.
+    **Arguments:**
+    - |drawCall|: The draw call parameters.
+    - |state|: The active [=RenderState=].
+
+    **Returns:** list of integer indices.
+
+    1. Let |vertexIndexList| be an empty list of indices.
+    1. If |drawCall| is an indexed draw call:
+        1. Initialize the |vertexIndexList| with |drawCall|.indexCount integers.
+        1. For |i| in range 0 .. |drawCall|.indexCount (non-inclusive):
+            1. Let |relativeVertexIndex| be [$fetch index$](|i| + |drawCall|.`firstIndex`,
+                |state|.[=RenderState/indexBuffer=].`buffer`, |state|.[=RenderState/indexBuffer=].`offset`,
+                |state|.[=RenderState/indexBuffer=].`format`).
+            1. If |relativeVertexIndex| is `"out of bounds"`, stop and return the empty list.
```

Comment:
why are we comparing the relative vertex index, and not `|drawCall|.`baseVertex` + |relativeVertexIndex|` instead?

Reply:
This is not comparing the vertex index, it's checking whether the variable `relativeVertexIndex` has the special value `"out of bounds"` which is generated by the `fetch index` algorithm. Clarified.